app/models/application_qualification.rb

```diff
@@ -144,6 +154,29 @@ def composite_equivalency_details
     details.strip if details.present?
   end
 
+  GCSE_PASS_GRADES = %w[A* A B C A*A* A*A AA AB BB BC CC CD 9 8 7 6 5 4 99 98 88 87 77 76 66 65 55 54 44 43].freeze
+  def failed_required_gcse?
+    return true if required_gcse? &&
+      all_grades.present? &&
```

Comment:
Super nitty, but i don't think we need to check for presence here

Reply:
I had a test failure without this, can't remember the test, but it came down to a qualification that didn't (yet) have a grade at all being flagged as failed. That doesn't make a lot of sense - you can't be flagged as failed if you've not entered a grade yet.